### PR TITLE
feat: implement docsite for contrib packages

### PR DESCRIPTION
**Pull request by dmytrokirpa**
To enhance the visibility of contrib packages, we've decided to create a documentation site that aggregates documentation for all packages in this repository, similar to what we do for the Fluent UI core repository. This will be integrated into `public-docsite-v9` using the Storybook composition feature.

**What's New:**
- Added a new `docsite` application powered by `storybook`.
- Selectively adopted Storybook styling and `DocsPage` components from the core to ensure a consistent user experience with `public-docsite-v9`, will be replaced with imports from a new  package once it's available (planned for Q3, cc @Hotell).
- Implemented a shared Storybook configuration in `/.storybook`.
- This shared configuration is now used across both apps and packages.
- Updated the titles and added basic descriptions to existing package stories, drawing from the package's README.md.

**Demo:**

https://github.com/user-attachments/assets/c915d299-14f4-4caa-a771-9a6adb694822

**Related Issues:**
Partially addresses issue #175.

### CODE REVIEW

File: .storybook/components/FluentDocsContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
```
Comment (dmytrokirpa) [inline .storybook/components/FluentDocsContainer.tsx:1]: This component was copied over from https://github.com/microsoft/fluentui/blob/master/apps/public-docsite-v9/src/DocsComponents/FluentDocsContainer.stories.tsx and will be removed once original component will be published as a part of a new package.

File: .storybook/components/FluentDocsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
```
Comment (dmytrokirpa) [inline .storybook/components/FluentDocsPage.tsx:1]: This component was copied over from https://github.com/microsoft/fluentui/blob/master/apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx and will be removed once original component will be published as a part of a new package.

File: .storybook/components/Toc.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
```
Comment (dmytrokirpa) [inline .storybook/components/Toc.tsx:1]: This component was copied over from https://github.com/microsoft/fluentui/blob/master/apps/public-docsite-v9/src/DocsComponents/Toc.stories.tsx and will be removed once original component will be published as a part of a new package.

File: .storybook/docs-root-v9.css [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/* remove the docs wrapper bg to let page bg show through */
```
Comment (dmytrokirpa) [inline .storybook/docs-root-v9.css:1]: This component was copied over from https://github.com/microsoft/fluentui/blob/master/apps/public-docsite-v9/.storybook/docs-root-v9.css and will be removed once original file will be published as a part of a new package.

File: .storybook/docs-root.css
Comment (dmytrokirpa) [inline .storybook/docs-root.css:1]: This component was copied over from https://github.com/microsoft/fluentui/blob/master/.storybook/docs-root.css and will be removed once original file will be published as a part of a new package.

File: apps/docsite/.storybook/theme.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { create } from '@storybook/theming';
```
Comment (dmytrokirpa) [inline apps/docsite/.storybook/theme.ts:1]: This component was copied over from https://github.com/microsoft/fluentui/blob/master/apps/public-docsite-v9/.storybook/theme.js and will be removed once original file will be published as a part of a new package.

File: apps/docsite/src/Welcome.mdx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+# Fluent UI Contributors Packages
+
+Welcome to the documentation site for Fluent UI Contributors packages! These packages are published under the `@fluentui-contrib/` scope and provide additional functionality to enhance your Fluent UI experience.
+
+## Available Packages
+
+Explore the various packages available to extend your Fluent UI capabilities:
+
+- [**houdini-utils**](/docs/packages-houdini-utils--docs): Utilities for CSS Houdini.
```
Comment (dmytrokirpa) [inline apps/docsite/src/Welcome.mdx:13]: Currently, the list is manually compiled, but we might explore generating it automatically from Storybook stories.

**Review by Jacqueline-ms — APPROVED** [review]
Azure theme looks good. Thanks for doing this!

File: apps/docsite/project.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  "projectType": "application",
+  "tags": [],
+  "targets": {
+    "lint": {
+      "executor": "@nx/eslint:lint"
+    },
+    "storybook": {
+      "executor": "@nx/storybook:storybook",
+      "options": {
+        "port": 4400,
```
Comment (mainframev) [inline apps/docsite/project.json:14]: should it start on the same port as storybook for stories? wont it be better to have different port for the docsite?

**Review by mainframev — CHANGES_REQUESTED** [review]
Great work on this! a few tiny things to improve 🚀

File: apps/docsite/.storybook/theme.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+  // Toolbar default and active colors
+  barSelectedColor: '#0078d4', // use msft primary blue default
+
+  // Form colors
+  inputBorderRadius: 4,
+
+  // Use the fluent branding for the upper left image
+  brandTitle: 'Fluent UI React Components',
+  brandUrl: 'https://github.com/microsoft/fluentui',
```
Comment (mainframev) [inline apps/docsite/.storybook/theme.ts:38]: as it was copied, I guess this part should be adjusted to contrib

```suggestion
  brandTitle: 'Fluent UI Contributors Packages',
  brandUrl: 'https://github.com/microsoft/fluentui-contrib',
```

File: apps/docsite/.storybook/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    "../**/*.spec.js",
+    "../**/*.spec.tsx",
+    "../**/*.spec.jsx"
+  ],
+  "include": [
+    "../stories/**/*.stories.ts",
+    "../stories/**/*.stories.js",
+    "../stories/**/*.stories.jsx",
+    "../stories/**/*.stories.tsx",
+    "../stories/**/*.stories.mdx",
```
[the remark this answers is not in the document]
Comment (dmytrokirpa) [reply apps/docsite/.storybook/tsconfig.json:23]: This file was autogenerated by nx, not sure we supposed to change it.

File: packages/react-tree-grid/stories/index.stories.tsx [hunk 10]
```diff
@@ -8,8 +9,15 @@ export { EmailInbox } from './EmailInbox.stories';
...
+  title: 'Packages/react-tree-grid',
   component: TreeGrid,
+  parameters: {
+    docs: {
+      description: {
+        component: description,
+      },
+    },
+  },
 } as Meta<typeof TreeGrid>;
```
[the remark this answers is not in the document]
Comment (dmytrokirpa) [reply packages/react-tree-grid/stories/index.stories.tsx:21]: I'd like to focus on making only the essential to the documentation site packages updates. Any other modifications should be managed by the respective owners.

File: apps/docsite/project.json (hunk 7 above)
Comment (dmytrokirpa) [reply apps/docsite/project.json:14]: I'd rather go with the default setup provided by NX instead of using a custom value.

File: apps/docsite/.storybook/theme.ts (hunk 8 above)
Comment (dmytrokirpa) [reply apps/docsite/.storybook/theme.ts]: done

**Review by mainframev — APPROVED** [review]

File: apps/docsite/project.json (hunk 7 above)
Comment (Hotell) [reply apps/docsite/project.json:14]: yup lets use default, it doesn't matter what project type we have, consistency wins

File: apps/docsite/src/Welcome.mdx (hunk 6 above)
Comment (Hotell) [reply apps/docsite/src/Welcome.mdx:13]: create issue to track this pls

File: .storybook/components/FluentDocsContainer.tsx (hunk 1 above)
Comment (Hotell) [reply .storybook/components/FluentDocsContainer.tsx:1]: can we aggregate these copy pasted things under a common folder ? `copied-from-fluentui-core/components` etc ?

more self explanatory then leaving comments per file I guess

also please open issue to track this TODO work. ty

File: .storybook/main.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import type { StorybookConfig } from '@storybook/react-webpack5';
```
Comment (Hotell) [inline .storybook/main.ts:1]: **note:** with this setup we are going against what we want ( following the general practices ) https://nx.dev/recipes/storybook/configuring-storybook 

> Nx does not generate a root Storybook directory and shared root Storybook configurations. Instead, it only generates a Storybook configuration for each project in your workspace. You may still manually create a root Storybook configuration file, if it is needed for your use case.


I know this is how we use it in core but that's not something we should do long term.

Anyways for consistency reasons we can keep this as is, and when we will start the migration in core we can then fast forward to contrib.

**changes needed:**

in order to properly trigger affected pipelines we need to manually adjust project graph ( another "magic" disadvantage of this setup ) 
see "inputs" https://github.com/microsoft/fluentui/blob/master/nx.json#L23


**the optimal solution we wont long term:**
- project level specific configs
- anything share-able lives in standard monorepo package which is used in scoped per project configs - proper API contract

File: packages/nx-plugin/src/generators/configure-storybook/files/.storybook/main.ts.template [hunk 12]
```diff
@@ -1,27 +1,11 @@
 import type { StorybookConfig } from '@storybook/react-webpack5';
 
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import rootConfig from '../../../.storybook/main';
```
Comment (Hotell) [inline packages/nx-plugin/src/generators/configure-storybook/files/.storybook/main.ts.template:4]: this wont scale, we need to use dynamic `offsetFromRoot` in order to generate valid paths https://nx.dev/nx-api/devkit/documents/offsetFromRoot

File: packages/nx-plugin/src/generators/configure-storybook/files/.storybook/preview.tsx.template [hunk 13]
```diff
@@ -1,18 +1,10 @@
+import type { Preview } from '@storybook/react';
 
-import * as React from 'react';
-
-import { Preview } from '@storybook/react';
-
-import { FluentProvider, webLightTheme } from '@fluentui/react-components';
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import rootPreview from '../../../.storybook/preview';
```
Comment (Hotell) [inline packages/nx-plugin/src/generators/configure-storybook/files/.storybook/preview.tsx.template:4]: same as above

File: packages/houdini-utils/README.md [hunk 14]
```diff
@@ -1,4 +1,4 @@
-# houdini-utils
+# @fluentui-contrib/houdini-utils
```
[the remark this answers is not in the document]
Comment (Hotell) [reply packages/houdini-utils/README.md:1]: if our generator doesn't add those lets fix that as well ( separate PR pls, this is big enough already )

File: packages/react-tree-grid/.storybook/tsconfig.json [hunk 15]
```diff
@@ -22,7 +22,6 @@
     "../stories/**/*.stories.tsx",
     "../stories/**/*.stories.mdx",
     "*.ts",
-    "*.js",
-    "../stories/VariantB.stories.tsx"
```
Comment (Hotell) [inline packages/react-tree-grid/.storybook/tsconfig.json:26]: whoops !

File: apps/docsite/src/Welcome.mdx (hunk 6 above)
Comment (dmytrokirpa) [reply apps/docsite/src/Welcome.mdx:13]: https://github.com/microsoft/fluentui-contrib/issues/246

File: packages/houdini-utils/README.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (dmytrokirpa) [reply packages/houdini-utils/README.md:1]: https://github.com/microsoft/fluentui-contrib/issues/247

File: packages/react-tree-grid/.storybook/tsconfig.json (hunk 15 above)
Comment (dmytrokirpa) [reply packages/react-tree-grid/.storybook/tsconfig.json]: reverted

File: packages/nx-plugin/src/generators/configure-storybook/files/.storybook/main.ts.template (hunk 12 above)
Comment (dmytrokirpa) [reply packages/nx-plugin/src/generators/configure-storybook/files/.storybook/main.ts.template]: done

File: .storybook/main.ts (hunk 11 above)
Comment (dmytrokirpa) [reply .storybook/main.ts:1]: Thank you for providing the background; it definitely clarifies things. We could potentially develop a local package with a storybook preset and integrate it across all contrib packages. However, I'm uncertain if it's beneficial since we'll eventually need to switch to the core version once it's available. In the meantime, as you recommended, I've added the "input" for the build-storybook  https://github.com/microsoft/fluentui-contrib/pull/244/files#diff-15552e50b1b7c05b05b7ffe08ee47b5ed62b8d2039229c58972a42fd22a7381cR25

File: packages/react-tree-grid/.storybook/tsconfig.json [hunk 16]
```diff
@@ -22,6 +22,7 @@
     "../stories/**/*.stories.tsx",
     "../stories/**/*.stories.mdx",
     "*.ts",
-    "*.js"
+    "*.js",
+    "../stories/VariantB.stories.tsx"
```
Comment (Hotell) [inline packages/react-tree-grid/.storybook/tsconfig.json]: I see my comment caused confusion :D sorry about that...

you initial change was correct. that "whoops" was my surprise that you exposed invalid setup , please revert this change to your former change. ty

**Review by Hotell — APPROVED** [review]
🚀

File: .github/CODEOWNERS [hunk 17]
```diff
@@ -1,3 +1,6 @@
+#### Applicaitions
```
Comment (layershifter) [inline .github/CODEOWNERS:1]: Typo?

**Review by layershifter — APPROVED** [review]

**Comment by Hotell** [comment]
admin merging
- pipeline passes
- got majority of approvals from owners
